src/power_grid_model/validation/validation.py

```diff
@@ -806,3 +809,20 @@ def validate_fault(data: SingleDataset) -> List[ValidationError]:
     errors += all_enabled_identical(data, "fault", "fault_type", "status")
     errors += all_enabled_identical(data, "fault", "fault_phase", "status")
     return errors
+
+
+def validate_transformer_tap_regulator(data: SingleDataset) -> List[ValidationError]:
+    errors = validate_base(data, "transformer_tap_regulator")
+    errors += all_valid_ids(
+        data,
+        "transformer_tap_regulator",
+        field="regulated_object",
+        ref_components=["transformer", "three_winding_transformer"],
+    )
+    errors += all_boolean(data, "transformer_tap_regulator", "status")
+    errors += all_valid_enum_values(data, "transformer_tap_regulator", "control_side", [BranchSide, Branch3Side])
+    errors += all_greater_than_or_equal_to_zero(data, "transformer_tap_regulator", "u_set")
+    errors += all_greater_than_zero(data, "transformer_tap_regulator", "u_band")
+    errors += all_greater_than_or_equal_to_zero(data, "transformer_tap_regulator", "line_drop_compensation_r")
+    errors += all_greater_than_or_equal_to_zero(data, "transformer_tap_regulator", "line_drop_compensation_x")
```

Comment:
```suggestion
    errors += all_greater_than_or_equal_to_zero(data, "transformer_tap_regulator", "line_drop_compensation_r", 0.0)
    errors += all_greater_than_or_equal_to_zero(data, "transformer_tap_regulator", "line_drop_compensation_x", 0.0)
```

Reply:
? why? the function is already called `all_greater_than_or_equal_to_zero`